### PR TITLE
[FLINK-17491] Translate Training page on project website

### PR DESCRIPTION
**Pull request by liying919**
Translate Training page on project website.

### CODE REVIEW

File: training.zh.md [hunk 1]
```diff
@@ -1,54 +1,52 @@
...
-title: "Training Course"
+title: "培训课程"
 ---
 
 <hr/>
 
-The Apache Flink community maintains a self-paced training course that contains
-a set of lessons and hands-on exercises. This step-by-step introduction to Flink focuses
-on learning how to use the DataStream API to meet the needs of common, real-world use cases.
+Apache Flink 社区维护了一套自学培训课程，包括了一系列课程和上手练习。 这个对 Flink 的逐步介绍关注于学习如何使用 DataStream API 来满足常见的实际使用需求。
```
Comment (klion26) [inline training.zh.md:7]: `这个对 Flink 的逐步介绍关注于学习如何使用 DataStream API 来满足常见的实际使用需求。` 这句话感觉有点拗口。能否修改一下呢

File: training.zh.md [hunk 2]
```diff
@@ -1,54 +1,52 @@
...
+                <span class="glyphicon glyphicon-random"></span> <b>DataStream API 简介</b>
             </div>
             <div class="panel-body">
                 <ul style="font-size: small;">
-                    <li>Data Types and Serialization</li>
-                    <li>Architecture</li>
-                    <li>Sources and Sinks</li>
+                    <li>数据类型和序列化</li>
+                    <li>架构</li>
+                    <li>数据源和 Sink</li>
```
Comment (klion26) [inline training.zh.md:35]: Source 和 sink 建议都继续翻译，和网站其他翻译保持一致

File: training.zh.md [hunk 3]
```diff
@@ -58,49 +56,49 @@ This training covers the fundamentals of Flink, including:
...
-                <span class="glyphicon glyphicon-time"></span> <b>Streaming Analytics</b>
+                <span class="glyphicon glyphicon-time"></span> <b>流式分析</b>
             </div>
             <div class="panel-body">
                 <ul style="font-size: small;">
-                <li>Event Time Processing</li>
-                <li>Watermarks</li>
-                <li>Windows</li>
+                <li>事件时间处理</li>
+                <li>水印</li>
```
Comment (klion26) [inline training.zh.md:64]: watermark 可以不翻译，可以参考 [wiki](https://cwiki.apache.org/confluence/display/FLINK/Flink+Translation+Specifications)

File: training.zh.md [hunk 4]
```diff
@@ -58,49 +56,49 @@ This training covers the fundamentals of Flink, including:
...
+                <span class="glyphicon glyphicon-ok"></span> <b>容错</b>
             </div>
             <div class="panel-body">
                 <ul style="font-size: small;">
-                <li>Checkpoints and Savepoints</li>
-                <li>Exactly-once vs. At-least-once</li>
-                <li>Exactly-once End-to-end</li>
+                <li>Checkpoints 和 Savepoints</li>
+                <li>精确一次与至少一次</li>
+                <li>精确一次的端到端</li>
```
Comment (klion26) [inline training.zh.md:93]: `端到端的精确一次` 会更好一些吗？

**Review by klion26 — COMMENTED** [review]
@liying919 thanks for your contribution, left some comments.

File: training.zh.md (hunk 1 above)
Comment (liying919) [reply training.zh.md:7]: 修改为了：
`这个对 Flink 的分步介绍重点在于学习如何使用 DataStream API 来满足常见的实际使用需求。`

File: training.zh.md (hunk 2 above)
Comment (liying919) [reply training.zh.md:35]: 参考了[wiki](https://cwiki.apache.org/confluence/display/FLINK/Flink+Translation+Specifications) 说sink不做翻译，这边source我也不翻译了吧

File: training.zh.md [hunk 5]
```diff
@@ -58,49 +57,49 @@ This training covers the fundamentals of Flink, including:
...
+                <li>端到端的精确一次</li>
                 </ul>
             </div>
         </div>
     </div>
 </div>
 
 <div style="margin-bottom: 400px;">
-<a href="{{site.DOCS_BASE_URL}}flink-docs-master/zh/training" target='_blank'>Apache Flink Training Course <small><span class="glyphicon glyphicon-new-window"></span></small> </a> 
+<a href="{{site.DOCS_BASE_URL}}flink-docs-master/zh/training" target='_blank'>Apache Flink 培训课程 <small><span class="glyphicon glyphicon-new-window"></span></small> </a>
```
Comment (klion26) [inline training.zh.md:102]: 这个链接会 404，可以参考英文版的继续下修改。

File: training.zh.md [hunk 6]
```diff
@@ -1,54 +1,53 @@
...
+                <span class="glyphicon glyphicon-copy"></span> <b>数据管道和 ETL</b>
             </div>
             <div class="panel-body">
                 <ul style="font-size: small;">
-                <li>Transformations</li>
-                <li>Stateful Stream Processing</li>
-                <li>Connected Streams</li>
+                <li>转换</li>
+                <li>有状态的流处理</li>
+                <li>关联的流</li>
```
Comment (klion26) [inline training.zh.md:49]: 这个如果改成`流的关联` 会好一些吗？

**Review by klion26 — COMMENTED** [review]
@liying919 thanks for the work. left another two comments. please take another look, thanks.

File: training.zh.md (hunk 6 above)
Comment (liying919) [reply training.zh.md:49]: 已修改

File: training.zh.md (hunk 5 above)
Comment (liying919) [reply training.zh.md:102]: 这里不太明白，英文版中的链接是
`href="{{site.DOCS_BASE_URL}}flink-docs-master/training" `
只差个 /zh/

Comment (klion26) [reply training.zh.md:102]: 英文版实际上是有跳转的（这个应该是在项目中某个地方设置了跳转，我暂时没有找到），最终的链接是 `https://ci.apache.org/projects/flink/flink-docs-master/learn-flink/index.html`

Comment (liying919) [reply training.zh.md:102]: @alpinegizmo Hi David, @klion26 is helping review this PR. We found the link _{{site.DOCS_BASE_URL}}flink-docs-master/training_ in training.md auto jumps to _https://ci.apache.org/projects/flink/flink-docs-master/learn-flink/index.html_ . Do you know where we can set the same jump config for Chinese version?

Comment (klion26) [reply training.zh.md:102]: I'm not very sure if we put the indeed URL(`https://ci.apache.org/projects/flink/flink-docs-master/learn-flink/index.html`) here is ok, currently, English version will redirect to it.

Comment (alpinegizmo) [reply training.zh.md:102]: @liying919 @klion26 That redirect is configured in docs/redirects/training.md.

**Comment by klion26** [comment]
hi @liying919,  what's the status of current PR?

**Comment by liying919** [comment]
> 
> 
> hi @liying919, what's the status of current PR?



> 
> 
> hi @liying919, what's the status of current PR?

The redirect config is in Apache/Flink.  @alpinegizmo Should we add the same redirect to https://ci.apache.org/projects/flink/flink-docs-master/zh/learn-flink/index.html in Project Flink?

**Comment by alpinegizmo** [comment]
@liying919 We needed that redirect in English because the old URL had already been published in a few places, and we didn't want to break those links. I don't know if the Chinese link has been published anywhere, but maybe not. 

Also, now that the training is present in the stable docs, we should not be linking to the training in the master docs.

File: training.zh.md (hunk 5 above)
Comment (alpinegizmo) [reply training.zh.md:102]: https://github.com/apache/flink-web/pull/357 updated this link to point to {{site.DOCS_BASE_URL}}flink-docs-stable/zh/learn-flink

**Comment by liying919** [comment]
@klion26 Please refer to the new commit.

**Comment by klion26** [comment]
@liying919 thanks for the work, the change LGTM, but there is some conflict, could you please rebase the master to resolve the conflict?

**Comment by liying919** [comment]
@klion26 The default branch has been rebased.

**Comment by klion26** [comment]
@liying919 thanks for the work. seems there is something wrong(now there are 258 files touched), could you please take another look.

**Comment by klion26** [comment]
@liying919 kindly ping.

**Comment by liying919** [comment]
> @liying919 thanks for the work, the change LGTM, but there is some conflict, could you please rebase the master to resolve the conflict?

Pls refer to the new PR.
https://github.com/apache/flink-web/pull/395

**Pull request closed without merging**